src/AbstractConsumer.php

```diff
@@ -256,8 +274,11 @@ protected function flushDeferred(): FlushDeferredResult
             $result = $callback($this->queue);
         } catch (\Throwable $e) {
             $this->logger->error('Exception during flushDeferred: ' . $e->getMessage());
-            $result = FlushDeferredResult::MSG_REJECT();
-            $this->handleException($e);
+            if ($this->handleException($e)) {
```

Comment:
I see, maybe handleDeveliveryException and handleFlushDeferredException, so we can return DeliveryResult or FlushDeferredResult directly and not true / false?

Reply:
We have the same error callback in both cases. The callback should return a value that can be interpret as DeliveryResult or FlushDeferredResult.

Or you suggest to split `handleException` on two methods and the callback may return boolean value?

Reply:
I see, you're right